tests/test_state.py

```diff
@@ -93,3 +94,43 @@ def test_state_serialize(
 
     # both states should have equivalent state, model parameters, loss, and outputs
     assert_state_equivalent(state1, state2)
+
+
+def test_state_batch_get_item():
+    state = get_dummy_state()
+    state.batch = [1, 2]
+    with patch('composer.core.state.batch_get') as mock_batch_get:
```

Comment:
nit: instead of using `unittest.mock.patch`, could you use the pytest `monkeypatch` fixture? This works; it's just that the rest of the codebase uses the monkeypatch fixture. See: https://docs.pytest.org/en/6.2.x/monkeypatch.html

Reply:
Ok, but I still have to use Mock to get the assert_called_with behavior.

Reply:
ok done here: 0164bf0